src/Nop.Plugin.Payments.SwedbankPayCheckout/SwedbankPayPaymentProcessor.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using Nop.Core;
+using Nop.Core.Domain.Orders;
+using Nop.Services.Configuration;
+using Nop.Services.Localization;
+using Nop.Services.Payments;
+using SwedbankPay.Sdk;
+using SwedbankPay.Sdk.PaymentOrders;
+using Nop.Services.Plugins;
+
+namespace Nop.Plugin.Payments.SwedbankPayCheckout
+{
+    public class SwedbankPayPaymentProcessor : BasePlugin, IPaymentMethod
+    {
+
+        private readonly IWebHelper webHelper;
+        private readonly IPaymentService paymentService;
+        private readonly SwedbankPayPaymentSettings swedbankPayPaymentSettings;
+        private readonly ISettingService settingService;
+        private readonly ILocalizationService localizationService;
+        private readonly ISwedbankPayClient swedbankPayClient;
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        public SwedbankPayPaymentProcessor(IPaymentService paymentService, ILocalizationService localizationService, IWebHelper webHelper, SwedbankPayPaymentSettings swedbankPayPaymentSettings, ISettingService settingService, ISwedbankPayClient swedbankPayClient, IHttpContextAccessor httpContextAccessor)
+        {
+            this.paymentService = paymentService;
+            this.localizationService = localizationService;
+            this.webHelper = webHelper;
+            this.swedbankPayPaymentSettings = swedbankPayPaymentSettings;
+            this.settingService = settingService;
+            this.swedbankPayClient = swedbankPayClient;
+            this.httpContextAccessor = httpContextAccessor;
+        }
+
+        public decimal GetAdditionalHandlingFee(IList<ShoppingCartItem> cart)
+        {
+            return paymentService.CalculateAdditionalFee(cart,
```

Comment:
All usage of private fields should be prefixed with `this.` for consistency.

```suggestion
            return this.paymentService.CalculateAdditionalFee(cart,
```